File: app/src/main/java/com/sanketvagadiya/ui/Projects/ProjectsFragment.java
package com.sanketvagadiya.ui.Projects;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProviders;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListAdapter;
import android.widget.ListView;

import com.sanketvagadiya.Constants.Constants;
import com.sanketvagadiya.Model.ProjectModel;
import com.sanketvagadiya.R;

import java.util.ArrayList;
import java.util.List;

public class ProjectsFragment extends Fragment {

    private ProjectsViewModel projectsViewModel;
    private Context mContext;
    private List<ProjectModel> projectList = new ArrayList<>();

    private ListAdapter adapter;
    private ListView lvProjectList;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        mContext = getActivity();
        projectsViewModel = ViewModelProviders.of(this).get(ProjectsViewModel
                .class);

        View root = inflater.inflate(R.layout.projects_fragment, container, false);

        lvProjectList = root.findViewById(R.id.lvProjectList);
        return root;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        if (Constants.isNetworkConnected(getContext())) {
            setUi();
        }
    }

    private void setUi() {
        projectList = projectsViewModel.getProjectList();
        if (projectList.size() > 0) {
            adapter = new ProjectListAdapter(mContext, projectList);
            lvProjectList.setAdapter(adapter);
        }
    }


}

File: app/src/main/java/com/sanketvagadiya/ui/Projects/ProjectsFragment.kt
package com.sanketvagadiya.ui.Projects

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProviders

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ListAdapter
import android.widget.ListView

import com.sanketvagadiya.Constants.Constants
import com.sanketvagadiya.Model.ProjectModel
import com.sanketvagadiya.R

import java.util.ArrayList

class ProjectsFragment : Fragment() {

    private var projectsViewModel: ProjectsViewModel? = null
    private var mContext: Context? = null
    private var projectList: List<ProjectModel> = ArrayList()

    private var adapter: ListAdapter? = null
    private var lvProjectList: ListView? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        mContext = activity
        projectsViewModel = ViewModelProviders.of(this).get(ProjectsViewModel::class.java)

        val root = inflater.inflate(R.layout.projects_fragment, container, false)

        lvProjectList = root.findViewById(R.id.lvProjectList)
        return root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        if (Constants.isNetworkConnected(context!!)) {
            setUi()
        }
    }

    private fun setUi() {
        projectList = projectsViewModel!!.projectList
        if (projectList.size > 0) {
            adapter = ProjectListAdapter(mContext!!, projectList)
            lvProjectList!!.adapter = adapter
        }
    }


}
